src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/VisualRuleForm.tsx

```diff
@@ -222,31 +225,53 @@ const VisualRuleForm: React.FC<IVisualRuleFormProps> = (props) => {
         []
     );
 
-    const onInternalModeChanged = useCallback((internalMode) => {
-        if (internalMode === 'INTELLISENSE') {
-            visualRuleFormDispatch({
-                type:
-                    VisualRuleFormActionType.FORM_VISUAL_RULE_EXPRESSION_TYPE_SET,
-                payload: { type: 'CategoricalValues' }
-            });
-            handleExpressionTextFieldEnabled(true);
-        } else {
-            visualRuleFormDispatch({
-                type:
-                    VisualRuleFormActionType.FORM_VISUAL_RULE_EXPRESSION_TYPE_SET,
-                payload: { type: 'NumericRange' }
-            });
-            handleExpressionTextFieldEnabled(false);
-        }
-    }, []);
+    const onInternalModeChanged = useCallback(
+        (internalMode) => {
+            if (internalMode === 'INTELLISENSE') {
+                visualRuleFormDispatch({
+                    type:
+                        VisualRuleFormActionType.FORM_VISUAL_RULE_EXPRESSION_TYPE_SET,
+                    payload: { type: 'CategoricalValues' }
+                });
+                handleExpressionTextFieldEnabled(true);
```

Comment:
what was the reasoning behind keeping a local state variable while also updating dispatch? Does it make sense to move this to the reducer since you're alwayss setting this value when you call that particular dispatch action?

Reply:
This is something the parent component needs to manage, since it is the component using this value to set styles correctly. This function is a prop.